psychopy/tools/filetools.py

```diff
@@ -241,3 +241,22 @@ def __del__(self):
             self.save()
         self._deleted = True
 
+def path_to_string(filepath):
+    """
+    Coerces pathlib Path objects to a string (only python version 3.6+)
+    any other objects passed to this function will be returned as is.
+    This WILL NOT work with on Python 3.4, 3.5 since the __fspath__ dunder
+    method did not exist in those verisions, however psychopy does not support
+    these versions of python anyways.
+     :Parameters:
+     filepath : string or pathlib Path object
+        file system path that needs to be coerced into a string to
+        use by Psychopy's internals
+     :Returns:
+    
+    filepath : string or same as passed object
+        file system path coerced into a string type
+    """
+    if hasattr(filepath, "__fspath__"):
+        return filepath.__fspath__()
+    return filepath
```

Comment:
There's a newline missing here :)

Reply:
Ah yes it should! I'll add another commit adding a new line and make sure that `path_to_string` is in the rest of the filetools functions.

For `tools.filetools.genFilenameFromDelimiter` should we work on ensuring that the function returns a pathlib object if a pathlib object is passed in as `filename` or should we always coerce it to return a string?

Reply:
I think ideally that would be the case; but then again it's only used internally anyway, I believe… So shouldn't really matter? I'm undecided!

Reply:
No strong opinion on that one. Yes, maybe simply return a string.